query/src/lib/base-query.ts

```diff
@@ -166,7 +166,7 @@ export function createBaseQuery<
             ...newOptions,
           },
           injector,
-        ) as QueryObserverOptions<
+        ) as unknown as QueryObserverOptions<
```

Comment:
@NetanelBasal dont know if this can be changed in a reasonable way, but it complained about an incompatibility on `TQueryKey` which suddenly was `TQueryKey & {}` (which is weird since its always a `ReadonlyArray<unknown>`, and this type only says that it should filter out null and undefined values - atleast this is what `& {}` does since i last checked)